airbyte-integrations/connectors/source-e2e-test/src/main/java/io/airbyte/integrations/source/e2e_test/InfiniteFeedSource.java

```diff
@@ -63,11 +64,13 @@ public AirbyteCatalog discover(final JsonNode config) {
       protected AirbyteMessage computeNext() {
         if (anotherRecordPredicate.test(i.get())) {
           if (i.get() != 0) {
-            try {
-              LOGGER.info("sleeping for {} ms", sleepTime);
-              sleep(sleepTime);
-            } catch (final InterruptedException e) {
-              throw new RuntimeException();
+            if (sleepTime.isPresent()) {
```

Comment:
nit: It could be refactor as
```
sleepTime.stream().foreach(st -> {
           try {
                LOGGER.info("sleeping for {} ms", st);
                sleep(st);
              } catch (final InterruptedException e) {
                throw new RuntimeException();
              }
})
```

Reply:
Using `stream().foreach()` on an optional feels a bit strange to me, as it seems to imply that it is iterating over a collection of objects. I personally feel like it is a bit more readable to just do the `if(sleepTime.isPresent())` check